frontend/src/metabase/redux/undo.js

```diff
@@ -76,7 +76,8 @@ export default function (state = [], { type, payload, error }) {
         _domId: previous._domId, // use original _domId so we don't get funky animations swapping for the new one
       });
     } else {
-      return state.concat(undo);
+      const currentUndoId = undo.id;
+      return state.filter(undo => undo.id !== currentUndoId).concat(undo);
```

Comment:
Prevent us from showing a new toast with the same `id`. There might be a better way than this though.